components/style/properties_and_values.rs

```diff
@@ -0,0 +1,1167 @@
+            if let Some(Syntax::Disjunction(ref mut ts)) = *syntax {
+                ts.push(Term { typ: t, list: false })
+            } else { unreachable!() }
+        }
+
+        /// Signal that we expect to be parsing some term in a disjunction.
+        fn expect_disjunction(syntax: &mut Option<Syntax>) {
+            if let Some(Syntax::Disjunction(_)) = *syntax {
+                // Good!
+            } else {
+                assert!(*syntax == None);
+                *syntax = Some(Syntax::Disjunction(vec![]))
+            }
+        }
+
+        /// Handle the next token in the syntax string (including whitespace).
+        fn handle_token(syntax: &mut Option<Syntax>, state: State, token: &Token) -> Result<State, ()> {
+            debug!("{:?} - {:?}", state, token);
+            match (state, token) {
+                (_, &Token::Comment(_)) => Err(()),
+
+                (State::Start { .. }, &Token::WhiteSpace(_)) => {
+                    // Ignore whitespace.
+                    Ok(State::Start { after_bar: false })
+                },
+                (State::Start { after_bar: false }, &Token::Delim('*')) => {
+                    // If we see a '*', that should be it (modulo whitespace).
+                    if syntax != &None {
+                        Err(())
+                    } else {
+                        *syntax = Some(Syntax::Anything);
+                        Ok(State::AfterAsterisk)
+                    }
+                },
+                (State::Start { .. }, &Token::Delim('<')) => {
+                    // A '<' should mean we're in the start of a '<type>'.
+                    expect_disjunction(syntax);
+                    Ok(State::AfterOpenAngle)
+                },
+                (State::Start { .. }, &Token::Ident(ref id)) => {
+                    // An identifier by itself should mean we're about to see a
+                    // specific identifier. Note that for <custom-idents> we
+                    // have that they "[must] not be misinterpreted as a
+                    // pre-defined keyword in that property’s value
+                    // definition". Here that means CSS-wide keywords!
+                    expect_disjunction(syntax);
+                    match CSSWideKeyword::from_ident(id) {
+                        Some(_) => Err(()), None => {
+                            push_type(syntax, Type::SpecificIdent(Ident((**id).into())));
+                            Ok(State::AfterType { after_whitespace: false })
+                        }
+                    }
+                },
+                (State::Start { .. }, _) => Err(()),
+
+                (State::AfterOpenAngle, &Token::Ident(ref id)) => {
+                    // We should be in something like '<length>' here.
+                    // https://drafts.css-houdini.org/css-properties-values-api/#supported-syntax-strings
+                    push_type(syntax, match &**id {
```

Comment:
Should this be ascii-case-sensitive? or insensitive? (need to go through the spec as a whole, it's pretty late here so I'm doing a first pass to get the overview of how this looks into my mind for tomorrow :P)

Reply:
ASCII-case-sensitive.

Reply:
Well, rather, codepoint matched (not familiar with ASCII-case-sensitive). https://github.com/w3c/css-houdini-drafts/issues/259#issuecomment-233049572